config/linkcheck-suppressions.txt

```diff
@@ -1339,3 +1339,6 @@
 <td><i><a href="com/puppycrawl/tools/checkstyle/xpath/iterators/ReverseListIterator.html#%3Cinit%3E(java.util.Collection)">com/puppycrawl/tools/checkstyle/xpath/iterators/ReverseListIterator.html#%3Cinit%3E(java.util.Collection)</a>: doesn't exist.</i></td></tr></table></td></tr>
 <td><i><a href="com/puppycrawl/tools/checkstyle/utils/UnmodifiableCollectionUtil.html#copyOfArray(T%5B%5D,int)">com/puppycrawl/tools/checkstyle/utils/UnmodifiableCollectionUtil.html#copyOfArray(T%5B%5D,int)</a>: doesn't exist.</i></td></tr>
 <td><i><a href="com/puppycrawl/tools/checkstyle/site/XdocsTemplateSink.CustomPrintWriter.html#%3Cinit%3E(java.io.Writer)">com/puppycrawl/tools/checkstyle/site/XdocsTemplateSink.CustomPrintWriter.html#%3Cinit%3E(java.io.Writer)</a>: doesn't exist.</i></td></tr>
+<td><i><a href="apidocs/com/puppycrawl/tools/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html">../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html</a>: doesn't exist.</i></td></tr>
```

Comment:
Why is this added?
Prove to me that link check is wrong and these links do exist.

Reply:
> Why is this added? Prove to me that link check is wrong and these links do exist.

to overcome the below CI failure
```
----------- grep of linkcheck.html--BEGIN
Checking internal (checkstyle website) links only.
35a36,38
> <td><i><a href="apidocs/com/puppycrawl/tools/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html">../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html</a>: doesn't exist.</i></td></tr>
> <td><i><a href="apidocs/com/puppycrawl/tools/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html">../../../../../com/puppycrawl/tools/checkstyle/checks/../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html</a>: doesn't exist.</i></td></tr>
> <td><i><a href="apidocs/com/puppycrawl/tools/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html">./com/puppycrawl/tools/checkstyle/checks/../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html</a>: doesn't exist.</i></td></tr>
```

Reply:
So then what is your reasoning that you think CI was NOT telling you something is wrong with your changes, and instead is a bug with link check?

Reply:
I guess the CI wasn't failing before my last push. It started failing on making these changes: `<td>Specify set of <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html">tokens</a> with limited occurrences as descendants.</td>`. But, fair point. When we click on the tokens, in the generated website, it redirects to the desired section. But, i do see the message in CI clearly. <a href="apidocs/com/puppycrawl/tools/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html">../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html</a>. Possibly it is doing some combination and giving links not valid.

Reply:
I recommend the first thing you do is figure out which of your changes are causing the failures. This will require you to try reverting stuff and running link check on your local. When you figure out which lines are the issue, tell us those lines, then generate that into whatever website is needed and test out the link and check it actually works.

If you need help let us know, but I can tell you, I can already tell you from what I saw, CI was not wrong about the failure.

Reply:
Hm,  generated website and all links are working. That is why I thought it is some limitation of our plugin execution.

Reply:
I believe I do, but I don't know for sure as I am not looking into it. If you want to see how I would tell if I am right or wrong, my instructions are at https://github.com/checkstyle/checkstyle/pull/13925#discussion_r1375309638 .

We shouldn't just assume CI is wrong. It seems like that is what is being done and it doesn't seem like anyone has looked into it to confirm that X link in Y file is being reported as wrong and it is shown to be wrong. If this is a bug with the utility, we should report it.

Reply:
@romani @relentless-pursuit Since this code is now being removed from the PR, the invalid links were coming from the Javadoc. Not the XDocs.

Reply:
I remember that there was some problem in apidoc 
https://github.com/checkstyle/checkstyle/pull/13176

Let's keep growing such suppression for now.